src/ja/common/message/base.py

```diff
@@ -3,33 +3,97 @@
 implemented in order for a class to be transferable between the components
 of JobAdder.
 """
-from abc import ABC
+from abc import ABC, abstractmethod
+from typing import Dict
 
 
 class Serializable(ABC):
     """
-    A base class for all objects which can be converted to and from a python
+    A base class for all objects which can be converted to and from a Python
     dictionary. The dictionary should contain only basic python types:
-    strings, ints, bytes, doubles, lists and dictionaries.
+    strings, ints, bytes, doubles, lists and dictionaries. The dictionary can
+    also be dumped as/read from a YAML string.
     """
 
+    @abstractmethod
+    def to_dict(self) -> Dict[str, object]:
+        """!
+        @return A dictionary representation of this object.The entries in the
```

Comment:
I'd suppose YAML doesn't support all kind of objects as values. Maybe this is a good place to describe what kind of objects are allowed? OTOH we'd need ints, bools, strings and lists/dictionaries with those types (and dictionaries of dictionaries of those types, etc.)

Reply:
It is literally impossible to define a finite type that does what we need.
The problem is that MyPy does not support recursive typing as of right now.
I'll explain on Monday.

Reply:
I guessed as much, but I meant more like describing what we expect from the dictionary in the comments.

Reply:
Sorry, I misread your comment.
You're right, we should define what is allowed.

Reply:
>I guessed as much, but I meant more like describing what we expect from the dictionary in the comments.

Didn't I already do that in the docstring of the Serializable class?

Reply:
oh yes, you did.